filters/shedder/admission_test.go

```diff
@@ -208,6 +240,127 @@ func TestAdmissionControl(t *testing.T) {
 	}
 }
 
+func TestAdmissionControlChainOnlyBackendErrorPass(t *testing.T) {
+	dm := metrics.Default
+	t.Cleanup(func() { metrics.Default = dm })
+	m := &metricstest.MockMetrics{}
+	metrics.Default = m
+	defer m.Close()
+
+	// backend with 50% error rate
+	var cnt int64
+	backend := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		i := atomic.AddInt64(&cnt, 1)
+		if i%2 == 0 {
+			w.WriteHeader(http.StatusInternalServerError)
+		} else {
+			w.WriteHeader(http.StatusOK)
+		}
+	}))
+	defer backend.Close()
+
+	spec := NewAdmissionControl(Options{}).(*AdmissionControlSpec)
+
+	argsLeaf := make([]interface{}, 0, 6)
+	argsLeaf = append(argsLeaf, "testmetric-leaf", "active", "5ms", 5, 5, 0.9, 0.95, 1.0)
+	fLeaf, err := spec.CreateFilter(argsLeaf)
+	if err != nil {
+		t.Fatalf("error creating leaf filter %q: %v", argsLeaf, err)
+		return
+	}
+	defer fLeaf.(*admissionControl).Close()
+
+	args := make([]interface{}, 0, 6)
+	args = append(args, "testmetric", "active", "50ms", 10, 10, 0.9, 0.95, 1.0)
+	f, err := spec.CreateFilter(args)
+	if err != nil {
+		t.Fatalf("error creating filter %q: %v", args, err)
+		return
+	}
+	defer f.(*admissionControl).Close()
+
+	fr := make(filters.Registry)
+	fr.Register(spec)
+
+	rLeaf := &eskip.Route{Filters: []*eskip.Filter{{Name: spec.Name(), Args: argsLeaf}}, Backend: backend.URL}
+	proxyLeaf := proxytest.New(fr, rLeaf)
+	defer proxyLeaf.Close()
+
+	r := &eskip.Route{Filters: []*eskip.Filter{{Name: spec.Name(), Args: args}}, Backend: proxyLeaf.URL}
+	proxy := proxytest.New(fr, r)
+	defer proxy.Close()
+
+	reqURL, err := url.Parse(proxy.URL)
+	if err != nil {
+		t.Fatalf("Failed to parse url %s: %v", proxy.URL, err)
+	}
+
+	rt := net.NewTransport(net.Options{})
+	defer rt.Close()
+
+	req, err := http.NewRequest("GET", reqURL.String(), nil)
+	if err != nil {
+		t.Error(err)
+		return
+	}
+
+	N := 1000
+	var wg sync.WaitGroup
+	wg.Add(N)
+	lim := rate.NewLimiter(50, 50)
+	now := time.Now()
+	for i := 0; i < N; i++ {
+		r := lim.Reserve()
+		for !r.OK() {
+			time.Sleep(10 * time.Microsecond)
```

Comment:
This makes tests run longer. We need to think whether it is necessary and maybe we can mock the clock like in https://github.com/zalando/skipper/pull/2432 and https://github.com/zalando/skipper/pull/2453

Reply:
I need to have something that jitters the funcs, because otherwise you get errors from the runtime.